azure-proto-core/ArmClientOptions.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+
+namespace azure_proto_core
+{
+    public class ArmClientOptions
+    {
+        private Dictionary<Type, object> _overrides = new Dictionary<Type, object>();
+        private static readonly object _overridesLock = new object();
+
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public object GetOverrideObject<T>(Func<object> ctor)
```

Comment:
Does this have to be public? Would this be used outside of extensions?

Reply:
It does have to be public its used by other DLLs as in the example of ArmClientOptionsExtensions.cs in azure-proto-compute

Reply:
Ic, that makes sense since it has to be use in ArmClientOptionsExtensions like classes which need to be static given our design. Kind of rules out inheritance or anything like that.